packages/website/bin/page-generator/write-pages.js

```diff
@@ -0,0 +1,230 @@
+const fse = require('fs-extra');
+const path = require('path');
+
+const {
+  changelogTemplate,
+  exampleTemplate,
+  singleComponentTemplate,
+  wrappedComponentTemplate,
+} = require('./templates');
+
+const writeFile = (pagePath, content) => {
+  fse.ensureFileSync(pagePath);
+  fse.writeFileSync(pagePath, content);
+};
+
+/**
+ * Generates an import path to a destination file relative from a given path
+ * @param from absolute path of the js file doing the import
+ * @param to absolute path of the file being imported
+ */
+const getImportPath = (from, to) => {
+  const fromDir = path.dirname(from);
+  if (!to) {
+    throw new Error('To path cannot be empty');
+  }
+  return path.relative(fromDir, to).replace('.js', '');
+};
+
+const getGenericPageInfo = (
+  pagesPath,
+  pagePath,
+  componentPath,
+  wrappersPath,
+  wrapperName,
+) => {
+  const absolutePagePath = path.resolve(pagesPath, pagePath);
+  const componentImportPath = componentPath
```

Comment:
Is it now impossible to be undefined?

Reply:
nope, if component path is missing then componentImportPath will be undefined